CONTRIBUTING.md

```diff
@@ -1,47 +1,297 @@
+Some examples of "Architecture" improvements:
 
-Please follow this style to make TiDB easy to review, maintain and develop.
+- Adding a new feature or making a feature more configurable/modular.
+- Converting structs to interfaces.
+- Improving test coverage.
+- Decoupling logic or creation of new utilities.
+- Making code more resilient (sleeps, backoffs, reducing flakiness, etc).
+
+These sorts of improvements are easily evaluated if they decrease lines of code without breaking functionality.
+
+If you are improving the quality of code, then justify/state exactly what you 'cleaning up' in your Pull Request so as
+not to waste reviewer time.
+
+If you're making code more resilient, test it locally to demonstrate how exactly your patch changes
+things.
+
+### Is this a performance improvement ?
+
+If you are submitting a performance bug, you MUST ALSO submit data that demonstrates your problem if you want the issue to
+remain open.  This can be done locally using go benchmark tests. Such improvements must come with benchmark test which convince reviewers it is 
+actually do better.
+
+Examples of how NOT to suggest a performance bug (these can really lead to a long review process and waste cycles):
+
+- We *should* be doing X instead of Y because it *might* lead to better performance.
+- Doing X instead of Y would reduce calls to Z.
+
+The above statements have basically no value to a reviewer, because neither is a strong, testable, assertive statement.
+This will land your PR in a no-man's-land zone (at best), or waste tons of time for a busy reviewer (at worst).
+
+Of course any improvement is welcome, but performance improvements are the hardest to review.  They often make code more
+complex, and to-often are not easily evaluated at review time due to lack of sufficient data submitted by the author
+of a performance improvement patch.
+
+Some examples of "Performance" improvements:
 
-### Format of the Commit Message
+- Improving a caching implementation.
+- Reducing calls to functions which are O(n^2), or reducing dependence on API server requests.
+- Changing the value of default parameters for proceeses, or making those values 'smarter'.
+- Parallelizing a calculation that needs to run on a large set of node/pod objects.
 
-We follow a rough convention for commit messages that is designed to answer two
-questions: what changed and why. The subject line should feature the what and
-the body of the commit should describe the why.
+These issues should always be submitted with (in decreasing order or value):
 
+- A golang Benchmark test.
+- benchmarkdb
+
+Without submitting data and results for your suggested performance improvements, its very possible that bikeshedding
+about meaningless possible performance optimizations could waste both reviewer time as well as your own.
+
+Since performance improvements deal with empirical systems, one playing in this space should be intimately familiar with
```

Comment:
the phrase "one playing in this space" is really strange, I think "practitioners" or other words/phrases would be better.